saleor/graphql/account/resolvers.py

```diff
@@ -45,6 +47,19 @@ def resolve_staff_users(info, query):
     return gql_optimizer.query(qs, info)
 
 
+def resolve_user(info, id):
+    requester = info.context.user or info.context.service_account
+    if requester:
+        _model, user_pk = graphene.Node.from_global_id(id)
+        if requester.has_perms(["account.manage_staff", "account.manage_users"]):
+            return models.User.objects.filter(pk=user_pk).first()
+        if requester.has_perm("account.manage_staff"):
+            return models.User.objects.filter(pk=user_pk, is_staff=True).first()
+        if requester.has_perm("account.manage_users"):
+            return models.User.objects.filter(pk=user_pk, is_staff=False).first()
```

Comment:
For returning customers we have the `customers` queryset. Bonus: it also returns staff users who are customers (they have any orders - this is allowed in Saleor). We need to make sure that if a staff users is at the same time a customer, and we're returning their data here (because I as requester have `manage_users` permission), no restricted staff-related fields are leaking.

Reply:
Again, `resolve_customers ` above should use the same queryset as this resolver.